features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/instrumentation/gestures/GesturesListener.kt

```diff
@@ -121,14 +121,68 @@ internal class GesturesListener(
 
     // region Internal
 
-    private fun findTargetForScroll(decorView: View, x: Float, y: Float): ViewTarget? {
-        return androidActionTrackingStrategy.findTargetForScroll(decorView, x, y)
-            ?: composeActionTrackingStrategy.findTargetForScroll(decorView, x, y)
+    private fun findTarget(
+        decorView: View,
+        x: Float,
+        y: Float,
+        isScroll: Boolean = false
+    ): ViewTarget? {
+        val queue = LinkedList<View>()
+        queue.addFirst(decorView)
+        var target: ViewTarget? = null
+
+        while (queue.isNotEmpty()) {
+            // removeFirst can't fail because we checked isNotEmpty
+            @Suppress("UnsafeThirdPartyFunctionCall")
+            val view = queue.removeFirst()
+            val newTarget = if (isScroll) {
+                findTargetForScroll(view, x, y)
+            } else {
+                findTargetForTap(view, x, y)
+            }
+            if (newTarget != null) {
+                target = newTarget
+            }
+            if (view.isVisible && view is ViewGroup) {
+                for (i in 0 until view.childCount) {
+                    val child = view.getChildAt(i)
+                    queue.add(child)
+                }
+            }
+        }
+
+        if (target == null) {
+            internalLogger.log(
+                InternalLogger.Level.INFO,
+                InternalLogger.Target.USER,
+                { MSG_NO_TARGET_TAP }
+            )
+        }
+        return target
     }
 
-    private fun findTargetForTap(decorView: View, x: Float, y: Float): ViewTarget? {
-        return androidActionTrackingStrategy.findTargetForTap(decorView, x, y)
-            ?: composeActionTrackingStrategy.findTargetForTap(decorView, x, y)
+    private fun findTargetForScroll(view: View, x: Float, y: Float): ViewTarget? {
+        // return most inside scrollable element
```

Comment:
is there a missing word here? same for the similar comment below

Reply:
I think that "most inside" is not the term to use in such case. Maybe "bottom-most" then? Because there is for sure "top-most" which exists.